scripts/dts/extract_dts_includes.py

```diff
@@ -732,78 +590,42 @@ def generate_include_file(defs, args):
             if prop == 'aliases':
                 for entry in sorted(defs[node][prop]):
                     a = defs[node][prop].get(entry)
-                    print_key_value(entry, a, maxtabstop)
+                    fd.write(get_key_value(entry, a, maxtabstop))
             else:
-                print_key_value(prop, defs[node].get(prop), maxtabstop)
+                fd.write(get_key_value(prop, defs[node].get(prop), maxtabstop))
 
-        sys.stdout.write("\n")
+        fd.write("\n")
 
-    if args.fixup:
-        for fixup in args.fixup:
+    if fixups:
+        for fixup in fixups:
             if os.path.exists(fixup):
-                sys.stdout.write("\n")
-                sys.stdout.write(
+                fd.write("\n")
+                fd.write(
                     "/* Following definitions fixup the generated include */\n")
                 try:
-                    with open(fixup, "r") as fd:
-                        for line in fd.readlines():
-                            sys.stdout.write(line)
-                        sys.stdout.write("\n")
+                    with open(fixup, "r") as fixup_fd:
+                        for line in fixup_fd.readlines():
+                            fd.write(line)
+                        fd.write("\n")
                 except:
                     raise Exception(
                         "Input file " + os.path.abspath(fixup) +
                         " does not exist.")
 
-    sys.stdout.write("#endif\n")
+    fd.write("#endif\n")
+    fd.close()
 
 
-def lookup_defs(defs, node, key):
-    if node not in defs:
-        return None
+def load_and_parse_dts(dts_file):
+    with open(dts_file, "r") as fd:
+        dts = parse_file(fd)
+        fd.close()
```

Comment:
The 'with' statement will close fd, so you don't have to explicitly close it here. This close() function call must be removed to not confuse the reader.